Tribes/src/app/progress-bar/progress-bar.component.ts

```diff
@@ -12,10 +13,17 @@ export class ProgressBarComponent implements OnInit {
   @Input() notification: Notification;
   result: number;
 
-  constructor() { }
+  constructor(private buildingService: BuildingService) {
+    this.buildingService.beginConstruction.subscribe({
+      next: () => {
+        this.refreshProgressBar();
+      }
```

Comment:
How often is this next happening? I have seen a discussion about something freezing the browser, this would be my candidate for a root cause.